### ISSUE TITLE
[Feature Request] Improved Consumables interface. 

### ISSUE THREAD

**Issue by ghost** [grokability/snipe-it#2534]
Hi,

We've just started using Snipe, it's been pretty useful thus far. Though managing consumables is a little finicky.

Ideally for our situation you should be able to check in new consumables when we receive them, rather than editing them manually. 

Thanks

**Comment by snipe** [comment]
I'm not sure I understand - can you describe your use-case a little more, and what you'd want to see?

**Comment by patrickc2o** [comment]
Yeah of course. 

We're a printing company so we tend to go through a lot of ink/consumables. Which means we are constantly buying more. 

The current way it works now is to add new consumables I have to either:

-  Manually edit the Consumable to a higher number which does not leave a trace of the edit or increase in stock.
- Create a new Consumable, which means tracking total usage of the particular Consumable is fairly awkward. 

Unless I'm missing it entirely (which is of course possible) it would be a lot easier to treat it like a stock system with being able to book in new products. Say having a book in button next to the check in button below, where the person booking in can input an order number if they would like. 

![image](https://cloud.githubusercontent.com/assets/21359521/18168232/a8e43e62-704c-11e6-9b10-47e559b4aca8.png)

Then when clicking on the consumable you would have a history of the amount used and amount coming in, something like the below. 
![image](https://cloud.githubusercontent.com/assets/21359521/18168444/958069f8-704d-11e6-94ec-cb0010fae902.png)

Hopefully that makes more sense.

**Comment by VanillaNinjaD** [comment]
Awesome explanation @patrickc2o, we are looking forward to this functionality as well.

It would streamline our inventory process of things like Copier/Printer Toner, and generic peripherals that we don't track serial numbers on.

**Comment by fltchr** [comment]
I'd like to see consumables work like this as well... I stock toner for a number of printers, some BW some CMYK, some with separate drums, others with maintenance kits. It would be very nice to have a restock interface.'

A plus for me would be for a "receiving" screen to load with all the consumables, set quantities received and have a notes field where I could put in an order number and see it logged somewhere that x consumables were replenished on order xyz on a particular date. That was you can receive multiple items at once all against the same order number.

**Comment by dexit** [comment]
I think SNIPE-IT would benefit from just a simple REPLENISH button for each Consumable/Component, it should include :
1) Incoming QTY
2) PO number
3) And PO date

**Comment by Oreios** [comment]
I'd like to add to Dexit's comment by adding the following.

4. Purchase cost
5. On the dashboard the total number represents the different kinds of consumables, not the total number (such as the other (eg licenses)
6. Check out to Asset (being a Printer itself) instead of user, or both.

**Comment by zdienos** [comment]
yeahh, I need this feature too

thank you

**Comment by mattgann** [comment]
Is there any documentation on any of the replenishment categories, Having total never change and seeing total is completely confusing to me. Does anyone have a simple explanation of any of the replenishment categories ?

**Comment by aviemet** [comment]
Are there any updates on this feature? Toner inventory management was actually the main reason I downloaded this to try out, the other asset management features were a bonus for me.

**Comment by raffo1987** [comment]
Any news about that?
It's very uncomfortable to add every new toner a new consumable anche the "solution" of increase manually the stock it's stupid because that way there is not trace.

Form me the best is use per product code every new supply for increase the stocking counter.

**Comment by DudeNamedBenFranklin** [comment]
I'll throw my request in here as well, we are looking for something to use for paper inventory, I need our staff to be able to add more paper by increasing the stock. We do this all the time. Thanks!

**Comment by edwardbattye** [comment]
I would also like this, would be good to have toner/standard peripherals like keyboard and mice and such. You just want an increase/decrease button for each item that would increase the stock value.

**Comment by stevenhorner** [comment]
I would also like to request this, we have exactly the same use case as @patrickc2o. We keep a stock of toners and need to update the stock level when we buy more. I thought I was missing something and found this issue while searching for documentation on using Consumables in Snipe-it. 

The current implementation just appears weird to me and not sure how you would actually use it without getting completely confused. Everything else in Snipe-it is excellent.

**Comment by kcarette** [comment]
Hi, I would also like seeing this functionality in consumables.  Just starting to use Snipe-It, which is a great tool, but the way the consumables work today is totally pointless to me.  As many others here do, we also manage our IT stock (toners, drums, ...) with snipe-it; but there is no option to add your stock.  So we should be able to increase our stock of toners with the amount delivered, a PO, ... so it's traceable.  Next to that, I also would love it when our end users could request consumables.  Requesting assets seems unneccessary to me, since the IT department decides when someone gets a new laptop or another asset; but being able as an end user to request a new toner (which automatically would lower the available stock by -1) would be great.  Since this feature / enhancement seems closed, I was wondering what the plans are for this.

**Comment by ReignSol** [comment]
Is the feature going to be opened again? This makes sense in many ways.

**Comment by headygains** [comment]
I want this feature so bad I'm almost willing to delve into the phpverse to make it happen.

**Comment by kcarette** [comment]
Since this functionality is not yet available within snipe-it; it's still relevant to me :-)

**Comment by jpcosta2** [comment]
Poking in as a new user - this is definitely relevant to me as well. I had hoped to utilize the consumables function at our printing company but without a quick way to replenish items for stock we go through quickly it's hard to justify using this. If consumables could be linked to a particular Model it would be especially useful.

**Comment by aviemet** [comment]
This should really be made a priority. Nobody is sitting on this page bumping the issue every time it becomes stale. The way consumeables are implemented now is pretty useless and the people in this thread have made a valid feature request which should really be implemented. We shouldn't need to continue posting the same complaints and requests just to keep this alive. Are there any plans to put some effort into making this feature useful?

**Comment by sabait** [comment]
Just Bumping this topic. I came here thinking I just couldn't find a way to add new stock to consumables, but found out this hasn't been implemented. Having a separate interface for re-stocking consumables would be great. Tracking prices would be a nice future improvement, but for now just being able to add toner with a PO number would be good enough. Bonus points if you can add multiple toners on the same interface by PO number or transaction.  Another idea would be to include a + sign on the actions column and which brings a window that asks PO (optional) and amount to be added (bonus points if cost included then averaged or FIFO). 

Also, is there an option to check out multiple toners at the same time? If a printer uses 4 toners for each color, then most likely the checkout would include multiple toners.

The idea of the users being able to requests toners would be amazing, but I guess it would be harder to implement. 

For now, the interface is not great to track inventory as it would be more practical to include a high number of total quantity and just track user requests. Having to manually edit each toner when you purchase 30 different toners at a time is kind of tedious. The only way to keep track of toner costs and PO at the moment would be to duplicate toners every time a purchase is done, which is not practical.

**Comment by edwardbattye** [comment]
Would still like to see this -it’s the main reason we aren’t fully making use of snipe it

**Comment by xotan135** [comment]
@snipe Is this feature still on the slate to become a reality someday?

**Comment by KimmoJ** [comment]
> 
> @snipe Is this feature request still on the slate to become a reality someday?

There's a more comprehensive rework of the consumables function linked to mid-thread, so that's hopeful. I sure hope this is being developed further.  As it stands, the consumables category is overly primitive, to say the least. But at least you can keep track of remaining inventory... that's not nothing.

**Comment by VanillaNinjaD** [comment]
Yes, AFAIK this is still as relevant as ever.

**Comment by xotan135** [comment]
> > @snipe Is this feature request still on the slate to become a reality someday?
> 
> There's a more comprehensive rework of the consumables function linked to mid-thread, so that's hopeful. I sure hope this is being developed further. As it stands, the consumables category is overly primitive, to say the least. But at least you can keep track of remaining inventory... that's not nothing.

Yeah I agree. I just would love to see this worked in at some point. I know that I have to manage expectations. We get this as an open source and free project, and in no way am I demanding this get put in and make it most important. It was just a curiosity question asking if this was still on their radar at some point.

**Comment by DudeNamedBenFranklin** [comment]
We would use this and promote it to other organizations if this was a
working feature. We constantly repair chromebooks and replace dongles and
HDMI cords and printer ink/toner.

If this was a robust feature, and could alert a secretary which PO to order
from it would be awesome. Until then, Google Sheets is what we are using.

As others have said, open source is great, but if this product worked well
for schools, we typically buy block hours or expert support contracts with
our open source software.

-- 
James Meyer

**Comment by xotan135** [comment]
Yeah the lack of this feature is the only thing keeping me from using it
right now.

The other things in snipe-it are great but I'm looking for an easy way to
manage consumables, and track them. Yeah I know I can edit an already
created consumable and put the number I bought in. But then I loose all the
overall usage stats I could have gain had it logged a replenish instead of
editing an already existing item.

That's just one example of consumables I have issue with.

**Comment by KimmoJ** [comment]
> Yeah I agree. I just would love to see this worked in at some point. I know that I have to manage expectations. We get this as an open source and free project, and in no way am I demanding this get put in and make it most important. It was just a curiosity question asking if this was still on their radar at some point.

Clearly more than one person finds this quite important, and I'm another. Plus, Snipe-it does offer cloud instances of the tool for money, and the more versatile the tool (while staying within its niche and not getting insane feature-itis of course) the bigger the chance of that, too.

**Comment by anhkernel** [comment]
I'm just learning SnipeIT right now and couldn't find anything in the user manual for how to use consumables. Based on the comments above my question is if what I have below is the only way to use Consumables?

1. Create new consumable - e.g. **Canon XYZ Toner**
2. Check out until depleted - e.g. **0/10**
3. Create new consumable again, even though it's the same exact item
4. Check out until depleted
5. Repeat

Is this right?

**Comment by xotan135** [comment]
Yep

**Comment by excedra** [comment]
> I'm just learning SnipeIT right now and couldn't find anything in the user manual for how to use consumables. Based on the comments above my question is if what I have below is the only way to use Consumables?
> 
> 1. Create new consumable - e.g. **Canon XYZ Toner**
> 2. Check out until depleted - e.g. **0/10**
> 3. Create new consumable again, even though it's the same exact item
> 4. Check out until depleted
> 5. Repeat
> 
> Is this right?

I'm not using it this way, because it spams the list of consumables. I'm increasing the sum of the consumable when I'm getting new items... 

I would like to see a "Add Items to Stock"-Button. Then it would be a very usefull feature!

**Comment by xotan135** [comment]
Agreed

**Comment by spotter-mmf** [comment]
+1

**Comment by tomck** [comment]
I am searching for asset and consumables management that makes sense for a small company, we have consumables like tape and mounts for installs and things we'll hopefully keep for a long time, like CPE and routers, and while I was happy that I could check out a consumable to a person, I couldn't see a way to mark the consumable as consumed. (We would want to know how many mounts or rolls of electrical tape a technician has, just to know if their install vehicle is stocked.)

I found [a Reddit thread where someone's considering using Snipe-IT](https://www.reddit.com/r/sysadmin/comments/chi8ci/pain_points_with_snipeit/), and decided against it for the lack of consumables ability to actually be consumed. I'm not here to be negative, however, it's a sign of a great project that the pressing issues are relatively minor.  Projects like this are really complex!

**Comment by xotan135** [comment]
It's been a feature request for a while bud.

The problem only really comes when you have to replenish said consumable
not consuming them.

Instead of just bringing in more on that line item you either have to edit
it and add more or add a whole new consumable line item.

It would also be nice if instead of checking out consumables the wording on
the button was a little different but that's relatively minor.

**Comment by Arapacana** [comment]
sigh... adding my bump.  Need this implemented or consumable tracking is essentially useless.  I really want to love snipe-it but this lack of basic feature implementation for years is maddening -- it doesn't inspire hope that these features will ever get added.

**Comment by xotan135** [comment]
Yeah I just read the brand new released hoping to read this as a new
feature but was sadly disappointed

**Comment by XionXinX** [comment]
Was eyeing Snipe-It but the lack of this feature is very deterring. Really hope it gets some attention.

**Comment by medwingADMIN** [comment]
Any news on this one? If I have toner and I want to have one spare and one int he printer, the only way to do it is to delete the toner and create a brand new one. It's crazy! I'm going back to a spreadsheet for now for toner, but love the rest of sniep

**Comment by xotan135** [comment]
> Any news on this one? If I have toner and I want to have one spare and one int he printer, the only way to do it is to delete the toner and create a brand new one. It's crazy! I'm going back to a spreadsheet for now for toner, but love the rest of sniep

yeah snipe is a great tool. I really wish they would add this feature because it would become usable for so many more people i think there have been two - three threads that talk about wanting this same thing. 

It got promoted to ready for dev around Mar - april 2019, and people keep asking for it but it has gotten little or no response from the dev team that i can tell. I really wish they would take notice of this feature request just because so many people have requested it.

**Comment by FairOrchid** [comment]
I'm disappointed to find that this much requested feature has gained no traction in four years ☹.

Not even a response either way as far as I can tell.

**Comment by Impartial-IT** [comment]
this would be very helpful.

**Comment by cryogenx** [comment]
@snipe any news on this one? the product is solid except for this... id be willing to send some budgetary dollars that way if it would help move the feature up the list.

**Comment by ruddens** [comment]
+1 on this please :)

**Comment by Solverz-0** [comment]
Is there plans for this feature to be added?

**Comment by xotan135** [comment]
Over the years I have following this thread. I have come to the conclusion
that they will adamantly ignore this thread until the last dev quits due to
old age

**Comment by xXValiXx** [comment]
+1 . Please add this feature because I don´t know how to use consumables the right way.

**Comment by ghost** [comment]
+1, this would be important to our team if we were to adopt snipe-it

**Comment by KimmoJ** [comment]
I'm honestly surprised nothing has happened in what, two years? Granted, the project chooses what they want to work on or not but still, it seems like a minor tweak at best that some would find quite useful. In our case, well... we've moved on to more automated ways of maintaining stock of things that might otherwise have been used here, so for us it's less of a priority now. Because we've moved on.

**Comment by ntaylor-86** [comment]
Just cloned the repo and looked into adding this feature.

![2022-06-02_15-52-09](https://user-images.githubusercontent.com/28693782/171562149-e1126812-2a32-43f6-9c62-efd593808638.jpg)

Fumbling my way through it 😵

Are the maintainers still up for adding this? 
What's the best place to ask for help this feature?

**Comment by ntaylor-86** [comment]
Alright, this is where I'm leaving it for tonight;

Replenish button on the Consumables page,
![2022-06-02_23-15-55](https://user-images.githubusercontent.com/28693782/171638316-cafb5622-101b-4937-8293-aa2fa2440222.jpg)

Replenish Consumable form,
![2022-06-02_23-16-42](https://user-images.githubusercontent.com/28693782/171638514-335c543f-41e5-4085-a949-b6beb0bbb3af.jpg)

Success Consumable replenished successfully,
![2022-06-02_23-17-05](https://user-images.githubusercontent.com/28693782/171638795-4231d303-9dcd-4bcb-b02d-6280ba1d6f7a.jpg)

**Comment by Solverz-0** [comment]
@ntaylor-86 This looks great, but I think it would work a lot better if instead of the "Total" increasing, the "remaining" should be increased instead.

For example, you know you will always want a total of no more than 10 reels of paper, you checkout 4 reels and now you have a remaining 6 reels. So you replenish this stock by ordering 4 reels and then you increase the remaining stock back up to 10 without affecting the total stock so the total still shows as 10 so you know how many max you should try to have.

**Comment by sabait** [comment]
I disagree, the total and remaining should be increased to know how many you have added in total. If you only increase remaining what happens if you add 10 when your  total is 10 and your remaining is 5? Total 10, remaining 15? Makes no sense. 
Max has little use for inventory purposes... if you do need to have that reference, you could add it as part of the name since it should never change.

To add to what is being worked on... I would add a Note for this form like the checkout has for consumables. That way you can include a PO number or any other information relevant to the increase in inventory.

**Comment by Solverz-0** [comment]
@sabait I see your reasoning, however no if you added 10 to the remaining when you total is 10 and your remaining is 5 then the total would be increased to 15 and the remaining increased to 10.

I disagree, max has little use for inventory purposes if you are buying and selling stock but for consumable stock it matters a lot as you do not want the person who orders the consumable stock to buy more than needed, only to top up the remianing to the total.

Tracking how many you have purchased over a certain period with the total is a feature I do not think is really relevent in an asset management system or should be an additional value. for example, total purchased and you can filter between dates etc.

**Comment by ghost** [comment]
@ntaylor-86 This is awesome, thank you for getting this implemented!

**Comment by landor** [comment]
Any chance there is any kind of audit / log of the replenish transaction
(who, when, quantity)?
Or is it just a shortcut for modifying the Quantity in the "Update
Consumable" screen? (Still awesome)
Thanks!

**Comment by xotan135** [comment]
Gaw I have not seen this much activity on this thread in years

**Comment by ntaylor-86** [comment]
> 

The logs of the replenish transaction is the next goal. I'll create a new table to store the transaction and display all of the the transactions on the individual consumable page

![2022-06-03_8-09-24](https://user-images.githubusercontent.com/28693782/171746793-32828ebb-6dc3-4917-9b17-a1fc2e2e4051.jpg)

**Comment by snipe** [comment]
I don't think we want to create new tables for that - it would go in the action logs table, so it can be reflected in the overall activity log.

**Comment by ntaylor-86** [comment]
Thanks @snipe I'd like to stick to conventions as much as possible!

**Comment by veenone** [comment]
Just want to let you know guys that I have PR #10856 about this topic pushed some times ago.
I think it's the same thing that being discussed here but with a little bit difference on the process.
Any feedback are welcomed

**Comment by ntaylor-86** [comment]
Looks good @veenone 👍

**Comment by veenone** [comment]
If you see something is missing in my pr, please let me know @ntaylor-86 🙂

**Comment by xotan135** [comment]
> > 
> 
> The logs of the replenish transaction is the next goal. I'll create a new table to store the transaction and display all of the the transactions on the individual consumable page
> 
> ![2022-06-03_8-09-24](https://user-images.githubusercontent.com/28693782/171746793-32828ebb-6dc3-4917-9b17-a1fc2e2e4051.jpg)

did your changes ever get merged ?

**Comment by ntaylor-86** [comment]
Hey @xotan135,

I put my development of this feature on hold after seeing @veenone already had such a detailed pull request in progress.

**Comment by xotan135** [comment]
gaw i thought i had seen in one of the patch notes they had added this
feature i guess i was mistaken. i pulled a docker container down to take a
look at it again and the replenish feature i still didn't see there.
